ml_project/tests/test_unittest.py

```diff
@@ -0,0 +1,176 @@
+import unittest
+from unittest.mock import patch
+import src
+import os
+import numpy as np
+import pandas as pd
+from numpy.random import randint
+
+from src.data import read_data, split_train_val_data
+from src.entities.predict_pipeline_params import (
+    PredictPipelineParams,
+    read_predict_pipeline_params,
+)
+from src.entities.train_pipeline_params import (
+    TrainingPipelineParams,
+    read_training_pipeline_params,
+)
+
+from src.features import make_features
+from src.features.build_features import extract_target, build_transformer
+from src.models import (
+    train_model,
+    serialize_model,
+    predict_model,
+    evaluate_model,
+)
+
+from src.models.model_fit_predict import create_inference_pipeline
+
+class DoesntExists(Exception):
+    print('File doesnt exist')
```

Comment:
use logger not print

Reply:
tricky, have added, not sure it works and that it is suitable to use logger in custom exception